### PR TITLE
Enum pretty-printing

### PR DESCRIPTION
**Pull request by psychon**
This supersedes #555 and depends on #556 (the commit from that PR is included in this one).

CC #554 and @andrewshadura 

------

    Improve Debug-printing of enum values
    
    When we switched from representing enums as Rust enums to representing
    them as newtypes around numbers, the derived Debug implementation
    followed and instead of printing pretty names, we got just numbers. This
    commit stops deriving a Debug implementation and instead one is
    implemented manually.
    
    The new Debug impl uses the alternate() flag to decide between
    UPPER_CASE and CamelCase. By default, the UPPER_CASE representation that
    is now also used in the source is used. With the alternate() flag,
    CamelCase is used instead.
    
    This also prints bitmasks as combinations of the individual values. For
    example, ModMask::SHIFT | ModMask::LOCK actually is represented as
    "SHIFT | LOCK". This was not possible before when we used a Rust enum to
    represent these.
    
    Signed-off-by: Uli Schlachter <psychon@znc.in>

### CODE REVIEW

**Comment by psychon** [comment]
Rebased on master

File: src/protocol/damage.rs [hunk 1]
```diff
@@ -88,6 +88,17 @@ impl From<u8> for ReportLevel {
...
 }
+impl std::fmt::Debug for ReportLevel  {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let variants = [
+            (Self::RAW_RECTANGLES.0.into(), "RAW_RECTANGLES", "RawRectangles"),
+            (Self::DELTA_RECTANGLES.0.into(), "DELTA_RECTANGLES", "DeltaRectangles"),
+            (Self::BOUNDING_BOX.0.into(), "BOUNDING_BOX", "BoundingBox"),
+            (Self::NON_EMPTY.0.into(), "NON_EMPTY", "NonEmpty"),
+        ];
+        pretty_print_enum(fmt, self.0.into(), &variants)
```
Comment (eduardosm) [inline src/protocol/damage.rs:99]: Why not a `match` (or two depending on `fmt`)?

Comment (psychon) [reply src/protocol/damage.rs:99]: Bad reason: I started writing the `bit` variant and then made the other version similar.

Slightly better reason, but I am not sure if this is true: #488. I hope that we only end up with one implementation of the code in the binary that just searches through some lookup table. With a `match`, I fear that the binary size would be larger. However, I didn't test this and it is just a theory.

Also: Two `match`es would mean more lines of code. :-)

**Review by eduardosm — APPROVED** [review]

**Comment by andrewshadura** [comment]
Yay, I have tested this and it works just as I wanted it :slightly_smiling_face:

**Pull request merged**